password_security/controllers/main.py

```diff
@@ -44,6 +45,7 @@ def web_login(self, *args, **kw):
             request.params['password']
         )
         if not uid:
+            request.uid = old_uid
```

Comment:
Oh and another good note - explicit is more Pythonic. Check out [PEP-20](https://www.python.org/dev/peps/pep-0020/):

> Explicit is better than implicit.